### DISCUSSION TITLE
Sticky key - trigger cmd-button after keypress or timeout?

### DISCUSSION THREAD

**Discussion by MarcCoquand** [kmonad/kmonad#928] [Q&A]
Heya, first and foremost thanks for creating this amazing tool.

I'd like to add a visual indicator that the sticky key is activated, for that as I understand it the correct tool is command buttons. I wrote the following, which triggers when I press lctl:
```
(defalias
   slc (around (cmd-button "echo 'sitcky-key on'") (sticky-key 700 lctl)))
```
What is less obvious to me is how to add something that triggers a cmd-button if I either A. I press something within 700 ms after pressing lctl B. 700 ms has passed and sticky-key is no longer active. 

Any pointers as to how I would implement something like that?

**Comment by jokesper** [comment]
I would have used a `tap-macro` instead of `around` (i.e.: `#((cmd-button "...")) (sticky-key 700 lctl)`)

For your question I would modify `lctl` to use a `tap-macro` (or maybe `around` with a `cmd-button`):
```clojure
#((cmd-button "X") (sticky-key 700 #((cmd-button "Y") lctl))
```
With this you have:
- On sticky (A): X, Y
- On timeout (B): X

So you would have X as a command, which waits for 700ms to detect if Y was run.
This means you would have to do some scripting to figure out what KMonad already nows.
Sadly this is (to my knowledge) the best (and only (?)) option, while we don't have an API.

EDIT:
Even better would probably implementing an optional `timeout-button` similar to `tap-hold-next-release`.
